resource_detectors/test/opentelemetry/detectors/deployment_test.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+describe OpenTelemetry::Resource::Detectors::Deployment do
+  let(:detector) { OpenTelemetry::Resource::Detectors::Deployment }
+
+  describe '.detect' do
+    let(:detected_resource) { detector.detect }
+    let(:detected_resource_attributes) { detected_resource.attribute_enumerator.to_h }
+    let(:expected_resource_attributes) { {} }
+
+    it 'returns an empty resource' do
+      _(detected_resource).must_be_instance_of(OpenTelemetry::SDK::Resources::Resource)
+      _(detected_resource_attributes).must_equal(expected_resource_attributes)
+    end
+
+    describe 'when in a rack environment' do
+      it 'returns a resource with rack environment' do
```

Comment:
There is a test helper [`with_env`](https://github.com/open-telemetry/opentelemetry-ruby/blob/50ce54bd7098c1bc280ec99b486a4f1f51f84ce7/common/lib/opentelemetry/common/test_helpers.rb#L7) that should be available for you to use for test cases.


Here are some examples: https://github.com/open-telemetry/opentelemetry-ruby/search?q=with_env

Reply:
Much shorter and cleaner, thanks for pointing me to this function :)
I refactored the code to use it